components/ProfileImageInfo.tsx

```diff
@@ -1,45 +1,163 @@
+      // Sentry.captureException(e)
+    }
+  }
+
   return (
     <div className="card shadow-sm">
       <div
-        className={`ms-auto me-auto mt-4 ${styles['profile_image_container']}`}
+        className={`ms-auto me-auto mt-4 mb-3 ${styles['profile_image_container']}`}
       >
         <UserInfoImage user={user} className="mt-4" />
       </div>
-      <h2 className="text-center mt-4">
+      <h4 className="text-center mt-4">
         {`${user.firstName} ${user.lastName}`}
-      </h2>
+      </h4>
       <div className="d-flex justify-content-center">
-        <h4 className="text-muted">{'@' + user.username}</h4>
+        <span className="fs-5 text-muted">{`@${user.username}`}</span>
       </div>
-      <div className="d-flex ms-auto me-auto mb-4">
+      <div className="d-flex ms-auto me-auto mb-4 mt-3 flex-column">
         {user.discordUserId ? (
           <>
-            <Image
-              src="/assets/discordClydeLogo.svg"
-              height={20}
-              width={20}
-              objectFit="contain"
-            />
-            <h5 className="ms-2">
-              <Link
-                href={`https://discordapp.com/users/${user.discordUserId}/`}
-              >
-                {user.discordUsername}
-              </Link>
-            </h5>
+            <div className="d-flex align-items-center mb-2">
+              <Image
+                src="/assets/discordClydeLogo.svg"
+                height={20}
+                width={20}
+                objectFit="contain"
+              />
+              <span className="fs-5 ms-2 mb-0">
+                <Link
+                  href={`https://discordapp.com/users/${user.discordUserId}/`}
+                >
+                  {user.discordUsername}
+                </Link>
+              </span>
+            </div>
+            <Button onClick={handleShow} type="danger" outline color="danger">
+              Unlink Discord
+            </Button>
+            <Modal show={show} onHide={handleClose}>
+              <ModalHeader closeButton>
+                <ModalTitle>Unlink Discord</ModalTitle>
+              </ModalHeader>
+              <ModalBody>
+                <p>
+                  You are about to unlink your Discord account. Please be aware
+                  that disconnecting your Discord account will degrade your
+                  learning experience in c0d3 as we use Discord as our community
+                  platform
+                </p>
+              </ModalBody>
+              <ModalFooter>
+                <BsButton variant="secondary" onClick={handleClose}>
+                  Close
+                </BsButton>
+                <BsButton
+                  variant="danger"
+                  onClick={handleUnlinkClick}
+                  data-testid="unlink-discord-btn"
+                >
+                  Unlink
+                </BsButton>
+              </ModalFooter>
+            </Modal>
+          </>
+        ) : (
```

Comment:
Just a general suggestion. Would it be a good idea to have the above(what's shown if user already has discord linked) and below(shown if user doesn't have discord linked), as their own separate components, then pass in the function and states like `show`, `handleUnlinkClick`, `handleConnectClick`, etc as props?

Reply:
Great idea, I'll update the code!